File: Java/StrategyPattern/src/Duck/WildDuck.java
package Duck;

import DuckBehaviours.Fly.IFlyBehaviour;
import DuckBehaviours.Quack.IQuackBehaviour;

public class WildDuck extends Duck{

    public WildDuck(IFlyBehaviour flyBehaviour, IQuackBehaviour quackBehaviour) {
        super(flyBehaviour, quackBehaviour);
    }

    public void performFly(){
        flyBehaviour.fly();
    }

    public void performQuack(){
        quackBehaviour.quack();
    }
}

File: Kotlin/StrategyPattern/src/ducks/WildDuck.kt
package ducks

import duckBehaviours.flyingBehaviour.IFlyingBehaviour
import duckBehaviours.quackBehaviour.IQuackBehaviour

class WildDuck(private val flyingBehaviour: IFlyingBehaviour, private val quackBehaviour: IQuackBehaviour) : Duck(flyingBehaviour,quackBehaviour) {

    override fun performFly(){
        flyingBehaviour.fly()
    }

    override fun performQuack() {
        quackBehaviour.quack()
    }
}